Feat [AccessControl]: Integrating with ExtremeXP-AccessControl

yagorezende opened the pull request.

yagosantosb2blue committed:
Feat [AccessControl]: Integrating with ExtremeXP-AccessControl

oralmpa committed:
Fix route handling in catalog.py to remove trailing slash

oralmpa reviewed (changes requested):
⚠️ Important Note:
The previous NGINX config explicitly set client_max_body_size 500M for file upload endpoints (e.g., /file/upload, /files/).
We have to add this directive in the new config, otherwise large uploads (over 1MB) will fail with HTTP 413 Request Entity Too Large.

Please make sure to include:

```nginx
client_max_body_size 500M;
```

in the file uploads  `location` blocks in the new NGINX config.

yagosantosb2blue committed:
Fix [NginX]: Fixing request body size limit

yagorezende commented:
> ⚠️ Important Note: The previous NGINX config explicitly set client_max_body_size 500M for file upload endpoints (e.g., /file/upload, /files/). We have to add this directive in the new config, otherwise large uploads (over 1MB) will fail with HTTP 413 Request Entity Too Large.
> 
> Please make sure to include:
> 
> ```nginx
> client_max_body_size 500M;
> ```
> 
> in the file uploads `location` blocks in the new NGINX config.

I changed the client_max_body_size to 4GB and removed the buffering limit restriction. This will fix the problem. Thanks for reporting.

oralmpa committed:
Log parsed catalog filter args for debugging proxy issues

oralmpa commented:
I’ve encountered a problem related to query parameters being stripped or misinterpreted when routed through the proxy.

When I bypass the proxy and hit the backend directly (via Postman or curl), the request works as expected:
`GET /ddm/catalog/list?file_type=json
`
The backend receives and processes the file_type=json parameter correctly.
However, when routed through the proxy, the backend logs show that file_type is missing, and request.args is empty.

This is likely due to the proxy_pass line in the NGINX config adding a ?, which causes double ? characters and breaks query parsing. I tried this:
`    location /ddm {
               proxy_pass http://proxy-service/proxy$uri$is_args$args;

               proxy_set_header Host $host;
               proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
               proxy_set_header X-Forwarded-Proto $scheme;
               proxy_buffering off;
               client_max_body_size 4096M;
          }
`
but i get 
`"OPTIONS /proxy/ddm/catalog/list?page=1&perPage=10&sort=created,desc HTTP/1.0" 404 - `

yagosantosb2blue committed:
Fix [Nginx]: Fixing query params parsing bug

yagosantosb2blue committed:
Merge branch 'feat-access-control-ys' of github.com:extremexp-HORIZON/DDM into feat-access-control-ys

oralmpa committed:
Merge main into feat-access-control-ys: resolved conflicts

oralmpa committed:
Refine datatable dark mode styles: remove button borders

oralmpa committed:
Remove redundant button styles from dropzone.css

oralmpa committed:
Fix: Exclude buttons from general dark-mode component background styling

oralmpa committed:
Enhance sidebar responsiveness on small screens

yagosantosb2blue committed:
Fix [NginX]: Bypassing proxy for swagger page
